packages/bot-web-ui/src/components/dashboard/dashboard.tsx

```diff
@@ -96,53 +99,76 @@ const Dashboard = ({
         }
     };
 
-    let getBotBuilderToken: string | number = '';
-    let getOnboardingToken: string | number = '';
+    const is_tour_complete = React.useRef(true);
+
+    let bot_tour_token: string | number = '';
+    let onboard_tour_token: string | number = '';
 
     React.useEffect(() => {
         if (active_tab === 0 && has_file_loaded) {
             onEntered();
         }
         if (active_tab === 0) {
-            getOnboardingToken = getTourSettings('token');
-            setOnBoardingTokenCheck(getOnboardingToken);
+            setTourType('onboard_tour_');
+            onboard_tour_token = getTourSettings('token');
+            setOnBoardingTokenCheck(onboard_tour_token);
         }
         if (active_tab === 1 && !has_onboard_tour_started) {
             setTourType('bot_builder_');
-            getBotBuilderToken = getTourSettings('token');
-            setBotBuilderTokenCheck(getBotBuilderToken);
-            setTourDialogVisibility(true);
+            bot_tour_token = getTourSettings('token');
+            setBotBuilderTokenCheck(bot_tour_token);
         }
-
         tour_status = getTourSettings('onboard_tour_status');
         setTourStatus(tour_status);
-    }, [active_tab, handleJoyrideCallback, has_onboard_tour_started]);
+    }, [active_tab, handleJoyrideCallback, has_onboard_tour_started, is_tour_complete, tour_status_ended]);
 
     //TODO: added addeventlistner because the useeffect does not trigger
-    window.addEventListener('storage', () => {
+    const botStorageSetting = () => {
         tour_status = getTourSettings('bot_builder_status');
         setTourStatus(tour_status);
-        getBotBuilderToken = getTourSettings('token');
+        if (tour_status_ended.key === 'finished') {
+            if (tour_type.key === 'onboard_tour_') {
+                setActiveTab(0);
+                setTourDialogVisibility(true);
+            } else {
+                setTourDialogVisibility(true);
+            }
+            setIsTourEnded(true);
+            is_tour_complete.current = false;
+            window.removeEventListener('storage', botStorageSetting);
+        }
+
+        bot_tour_token = getTourSettings('token');
         if (active_tab === 1 && !storage.bot_builder_token && !has_onboard_tour_started) {
             setTourSettings(new Date().getTime(), `${tour_type.key}token`);
         }
-    });
-    const token = false;
+    };
+    if (!bot_tour_token) {
+        setIsTourEnded(false);
+        window.addEventListener('storage', botStorageSetting);
+    }
+
     let storage = '';
```

Comment:
lets move all the variable declarations at the top of the component or before the methods/hooks
@farabi-deriv

Reply:
done ✅